pkg/ravencontroller/controllers/gateway_controller.go

```diff
@@ -238,6 +245,30 @@ func getNodeInternalIP(node corev1.Node) string {
 	return ip
 }
 
+// getPodCIDR returns the pod IP range assigned to the node.
+func (r *GatewayReconciler) getPodCIDR(ctx context.Context, node corev1.Node) (string, error) {
+	podCIDR := node.Spec.PodCIDR
+	for key := range node.Annotations {
+		if strings.Contains(key, "projectcalico.org") {
+			var blockAffinityList calicov3.BlockAffinityList
+			err := r.List(ctx, &blockAffinityList)
+			if err != nil {
+				err = fmt.Errorf("unable to list calico blockaffinity: %s", err)
+				return "", err
+			}
+
+			for _, v := range blockAffinityList.Items {
+				if v.Spec.Node != node.Name || v.Spec.State != "confirmed" {
+					continue
+				}
+				podCIDR = v.Spec.CIDR
```

Comment:
It is possible for nodes in calico to have multiple cidr block, so there can be multiple `blockaffinities` indicating the same node.
Currently, the Gateway CRD cannot handle multiple podCIDR circumstances well(`NodeInfo.Subnet` is a string).  I think we may add a subnet slice to `Gateway` CR to handle this, and the `raven-agent` also needs some adjustments.

Reply:
@DrmagicE You are right. I have realized it too when adding calico cidr block. As you mentioned that `raven-agent` and gateway CR must be adjusted, so I just get one cidr block to adapt current case. When `raven-agent` supported multiple cidr block, I will adjust calico to adapt it again.

Reply:
how about changing `NodeInfo.Subnet`(string) to `NodeInfo.Subnets`(slice)

Reply:
@njucjc good idea, agree.

Reply:
> 

@luckymrwang Can you change it together in this PR?